### PR TITLE
Added Front end Checkout Page and Error Page, Added Back end Checkout Feature, Updated Doc, Fixed bug#30

### PR DESCRIPTION
**Pull request by special-PPT**
# Pull Request Template

## Description
Please provide a detailed description of your pull request. Include its purpose, what problem it solves, etc.

## Type of Change
Please check the options that are relevant to your PR:
- [x] Bug fix
- [x] New feature
- [x] Documentation update
- [x] Code refactor
- [ ] Other, please describe:

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes:
1. 
2. 
3. 

## Checklist
Before submitting your PR, please make sure you have completed the following steps:
- [x] I have tested my changes locally and they do not break existing functionality.
- [x] I have updated the documentation accordingly.
- [x] I follow the code style of this project.

## Additional Information
If applicable, add any other information that could help the reviewers.

### CODE REVIEW

File: src/back-end/routes/cart.js
Comment (yanruc123) [inline src/back-end/routes/cart.js:1]: I will add route protection for this file later

File: src/back-end/routes/order.js
Comment (yanruc123) [inline src/back-end/routes/order.js:1]: I will add route protection for this file later

File: src/front-end/src/App.js
Comment (yanruc123) [inline src/front-end/src/App.js:1]: I will add route protection for this file later

**Review by yanruc123 — APPROVED** [review]

File: src/back-end/routes/cart.js
Comment (special-PPT) [reply src/back-end/routes/cart.js:1]: Thanks

File: src/back-end/routes/order.js
Comment (special-PPT) [reply src/back-end/routes/order.js:1]: Thanks

File: src/front-end/src/App.js
Comment (special-PPT) [reply src/front-end/src/App.js:1]: Thanks

**Pull request merged**